thorlcr/graph/thgraphmaster.cpp

```diff
@@ -1618,17 +1618,18 @@ void CJobMaster::broadcast(ICommunicator &comm, CMessageBuffer &msg, mptag_t mpt
             unsigned timeout;
             StringAttr errorMsg;
             ICommunicator &comm;
+            bool skipDisconnected;
         public:
-            CSendAsyncfor(ICommunicator &_comm, CMessageBuffer &_msg, mptag_t _mptag, unsigned _timeout, const char *_errorMsg)
-                : comm(_comm), msg(_msg), mptag(_mptag), timeout(_timeout), errorMsg(_errorMsg)
+            CSendAsyncfor(ICommunicator &_comm, CMessageBuffer &_msg, mptag_t _mptag, unsigned _timeout, const char *_errorMsg, bool _skipDisconnected)
+                : comm(_comm), msg(_msg), mptag(_mptag), timeout(_timeout), errorMsg(_errorMsg), skipDisconnected(_skipDisconnected)
             {
             }
             void Do(unsigned i)
             {
-                if (!comm.send(msg, i+1, mptag, timeout))
+                if (!comm.send(msg, i+1, mptag, timeout, skipDisconnected))
```

Comment:
see comment in mpcomm.
I think it would be better to handle the special case here, with code like:
```
                if (comm.verifyConnection(i+1, (unsigned)-1, !aborting))
                {
                    if (!comm.send(msg, i+1, mptag, timeout))
                        throw createBCastException(i+1, errorMsg);
                }
```

where verifyConnection has an option to not reconnect (NB: the CMPChannel::verifyConnection already does have this option).

Reply:
Btw, rather than extending CSendAsyncfor with all the members/parameters that are needed,
using a std function with [&] capture instead with std::async and a std::future will simplify the code quite a lot.

Reply:
yes, I'm fine with the broadcastSendAsync conditional being removed, it was only put/left in when the change was made, as a failsafe if we needed to turn it off, but it's been there now for a long time.